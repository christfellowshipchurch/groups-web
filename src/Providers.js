import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
