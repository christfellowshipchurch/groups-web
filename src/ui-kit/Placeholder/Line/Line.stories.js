import React from 'react';
import { storiesOf } from '@storybook/react';

import Line from '.';

storiesOf('ui-kit/Placeholder/Line', module)
  .addDecorator((story) => (
    <div style={{ width: '50vw', height: '50vw' }}>{story()}</div>
  ))
  .add('default', () => <Line>Renderd child</Line>)
  .add('isLoading ', () => <Line isLoading />)
  .add('style', () => <Line isLoading style={{ background: 'salmon' }} />)
  .add('width', () => <Line isLoading width={'50%'} />)
  .add('showLoadingAnimation', () => <Line isLoading showLoadingAnimation />);
