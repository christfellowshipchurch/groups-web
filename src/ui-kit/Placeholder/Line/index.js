import React from 'react';
import styled from 'styled-components';
import ReactPlaceholder from 'react-placeholder';
import PropTypes from 'prop-types';

const StyledLine = styled.mark.attrs({
  // `custom-placeholder` class enables the `showLoadingAnimation` prop to work
  className: 'ui-kit-Placeholder-Line custom-placeholder',
})`
  background: ${({ theme }) => theme.colors.background.inactive};
  border-radius: ${({ theme }) => theme.sizing.baseBorderRadius};
  display: block;
  height: 1rem;
  width: ${({ customWidth }) => customWidth};

  ${({ customStyles }) => customStyles}
`;

const Line = ({ isLoading, style, width, ...props }) => (
  <ReactPlaceholder
    customPlaceholder={<StyledLine customStyles={style} customWidth={width} />}
    ready={!isLoading}
    {...props}
  />
);

Line.defaultProps = {
  isLoading: true,
  width: '100%',
};

Line.propTypes = {
  isLoading: PropTypes.bool,
  style: PropTypes.any, // eslint-disable-line react/forbid-prop-types
  width: PropTypes.string,
};

Line.displayName = 'ui-kit.Placeholder.Line';

export default Line;
