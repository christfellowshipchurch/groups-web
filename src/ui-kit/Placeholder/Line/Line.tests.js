import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../../Providers';

import Line from '.';

describe('the Line component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <Line />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a custom width', () => {
    const tree = renderer.create(
      <Providers>
        <Line width={'50%'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
