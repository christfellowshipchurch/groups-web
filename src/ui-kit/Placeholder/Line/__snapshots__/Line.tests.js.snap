// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Line component should render 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

<mark
  className="c0 ui-kit-Placeholder-Line custom-placeholder"
/>
`;

exports[`the Line component should render children when isLoading is false 1`] = `"Rendered Child"`;

exports[`the Line component should render showLoadingAnimation 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

<mark
  className="c0 show-loading-animation ui-kit-Placeholder-Line custom-placeholder"
/>
`;

exports[`the Line component should render with a custom style 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
  background: salmon;
}

<mark
  className="c0 ui-kit-Placeholder-Line custom-placeholder"
/>
`;

exports[`the Line component should render with a custom width 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 50%;
}

<mark
  className="c0 ui-kit-Placeholder-Line custom-placeholder"
/>
`;
